control/encode_test.go

```diff
@@ -0,0 +1,100 @@
+package control_test
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"pault.ag/go/debian/control"
+	"pault.ag/go/debian/dependency"
+	"pault.ag/go/debian/version"
+)
+
+type TestMarshalStruct struct {
+	Foo string
+}
+
+type SomeComplexStruct struct {
+	control.Paragraph
+
+	Version    version.Version
+	Dependency dependency.Dependency
+}
+
+type TestParaMarshalStruct struct {
+	control.Paragraph
+	Foo string
+}
+
+func TestExtraMarshal(t *testing.T) {
+	el := TestParaMarshalStruct{}
+
+	isok(t, control.Unmarshal(&el, strings.NewReader(`Foo: test
+X-A-Test: Foo
+`)))
+
+	assert(t, el.Foo == "test")
+
+	writer := bytes.Buffer{}
+	isok(t, control.Marshal(&writer, el))
+	assert(t, writer.String() == `Foo: test
+X-A-Test: Foo
+
+`)
```

Comment:
Any particular reason this comes with extra whitespace at the end when it's just a single paragraph?